parse/ast.go

```diff
@@ -335,9 +330,10 @@ func (ae AnnotationExpression) validate() error {
 
 type QuotedLiteral string
 
-func (QuotedLiteral) node()    {}
-func (QuotedLiteral) literal() {}
-func (QuotedLiteral) value()   {}
+type NameLiteral string
+
+type NumberLiteral float64
```

Comment:
shouldn't we support all numeric types?

Reply:
Only allowed to pass in float64 as function argument?
What about supporting other types like int, unit, complex etc and approximations?

Reply:
its not the scope of this PR, as it was not changed only moved.

but you can cast int, unit, exponent, to float64, with no problems.
The only problem is complex numbers

Reply:
I guess add a comment :)

Reply:
complex numbers are not supported in MF2, the format is defined as JSON number https://www.rfc-editor.org/rfc/rfc8259#section-6

Reply:
yeah seems correct, so all other types are safely can be converted to float64, so no problemo